lib/ssl/src/tls_gen_connection.erl

```diff
@@ -389,27 +389,18 @@ handle_protocol_record(#ssl_tls{type = ?APPLICATION_DATA, fragment = Data}, Stat
             next_event(StateName, Record, State)
     end;
 %%% TLS record protocol level handshake messages 
-handle_protocol_record(#ssl_tls{type = ?HANDSHAKE, fragment = Data}, 
-		    StateName, #state{protocol_buffers =
-					  #protocol_buffers{tls_handshake_buffer = Buf0} = Buffers,
-                                      connection_env = #connection_env{negotiated_version = Version},
-                                      static_env = #static_env{role = Role},
-				      ssl_options = Options} = State0) ->
+handle_protocol_record(#ssl_tls{type = ?HANDSHAKE, fragment = Data}, StateName,
+                       #state{ssl_options = Options, protocol_buffers = Buffers} = State0) ->
     try
-	%% Calculate the effective version that should be used when decoding an incoming handshake
-	%% message.
-	EffectiveVersion = effective_version(Version, Options, Role, StateName),
-        {Packets, Buf} = tls_handshake:get_tls_handshakes(EffectiveVersion,Data,Buf0, Options),
-
-        State = case EffectiveVersion =/= NegotiatedVersion of
-                    State0#state{protocol_buffers =
-                                     Buffers#protocol_buffers{tls_handshake_buffer = Buf}},
+        {Packets, Rest, RecRest} = get_tls_handshakes(Data, StateName, State0),
+            State = State0#state{protocol_buffers = Buffers#protocol_buffers{tls_handshake_buffer = Rest}},
 	case Packets of
             [] -> 
-                assert_buffer_sanity(Buf, Options),
+                #protocol_buffers{tls_handshake_buffer = HSBuffer} = State#state.protocol_buffers,
```

Comment:
I think the variable `Rest`from line 395 already has the value so no need to dig it out from `State` again.
This line can be omitted if the variable `Rest` is renamed to `HSBuffer` or maybe this line could just be changed to `HSBuffer = Rest`